src/audio/google/google_rtc_audio_processing.c

```diff
@@ -632,6 +632,26 @@ static int google_rtc_audio_processing_prepare(struct processing_module *mod,
 	return 0;
 }
 
+static int trigger_handler(struct processing_module *mod, int cmd)
+{
+	struct google_rtc_audio_processing_comp_data *cd = module_get_private_data(mod);
+
+	/* Ignore and halt propagation if we get a trigger from the
+	 * playback pipeline: not for us.
+	 */
+	if (cd->ref_comp_buffer->walking)
+		return PPL_STATUS_PATH_STOP;
```

Comment:
IIUC (not sure about that) PPL_STATUS_PATH_STOP is kind of error code during propagation, which will be delivered towards upstream to negate the event through the whole pipeline.

That is, when the playback pipeline (as refpath) is triggered an event by the client, returning PPL_STATUS_PATH_STOP could make the event non-executed through the triggered pipeline. But I think what the client expects is to execute the event along the playback pipeline.

I wonder if we should return 0 here instead.

Reply:
Zero is a success condition, if I return that then what happens is that the pipeline walker continues along the capture path resetting (or whatever) the capture components incorrectly.   PATH_STOP seems like the right choice here, though I'll admit I don't think there's a single correct answer for pipeline management in IPC3.  Note that in IPC4 these never propagate, maybe I should gate the test on IPC_MAJOR_3 for clarity.